vmm/src/vm.rs

```diff
@@ -920,6 +940,42 @@ impl Vm {
         Ok(cmdline)
     }
 
+    #[cfg(target_arch = "aarch64")]
+    #[allow(deprecated)]
+    fn load_kernel_to_mem(
+        &mut self,
+        guest_mem: &GuestMemoryMmap,
+        guest_addr: GuestAddress,
+    ) -> Result<KernelInfo> {
+        let mut kernel_image = self.kernel.as_ref().unwrap();
+        let kernel_size = kernel_image
+            .seek(SeekFrom::End(0))
+            .map_err(|_| Error::SeekKernelEnd)? as usize;
+
+        if kernel_size > arch::aarch64::layout::KERNEL_IMAGE_MAX_SIZE as usize {
+            return Err(Error::KernelTooBig);
+        }
+
+        kernel_image
+            .seek(SeekFrom::Start(0))
+            .map_err(|_| Error::SeekKernelStart)?;
+
+        let kernel_addr = if guest_addr.0 == 0 {
+            arch::initramfs_load_addr(guest_mem, kernel_size).map_err(|_| Error::KernelLoadMem)?
+        } else {
+            guest_addr.0
+        };
+
+        guest_mem
+            .read_exact_from(GuestAddress(kernel_addr), &mut kernel_image, kernel_size)
```

Comment:
If this is why you have to use `#[allow(deprecated)]`, you can use its alternative `read_exact_volatile_from`. And this commit is a good reference: https://github.com/cloud-hypervisor/cloud-hypervisor/commit/d4892f41b3124e777724ed872fb9997b8a47c1fb